detect.go

```diff
@@ -1,70 +1,196 @@
-			return packit.DetectResult{}, packit.Fail
+		if config.Path == "" && len(projectFiles) == 0 {
+			return packit.DetectResult{}, packit.Fail.WithMessage("no *.runtimeconfig.json or project file found")
 		}
 
-		return packit.DetectResult{
-			Plan: packit.BuildPlan{
-				Provides: []packit.BuildPlanProvision{
-					{
-						Name: "dotnet-execute",
-					},
+		if len(projectFiles) > 0 {
+			projectFile := projectFiles[0]
+			version, err := projectParser.ParseVersion(projectFile)
+			if err != nil {
+				return packit.DetectResult{}, err
+			}
+
+			requirements = append(requirements, packit.BuildPlanRequirement{
+				// later todo: change to dotnet-application after dotnet-publish cnb is done
+				Name: "build",
+				Metadata: map[string]interface{}{
+					"launch": true,
 				},
-				Requires: []packit.BuildPlanRequirement{
-					{
-						Name: "dotnet-execute",
-						Metadata: map[string]interface{}{
-							"build": true,
-						},
+			})
+
+			requirements = append(requirements, packit.BuildPlanRequirement{
+				Name: "dotnet-runtime",
+				Metadata: map[string]interface{}{
+					"version":        version,
+					"version-source": filepath.Base(projectFile),
+					"launch":         true,
+				},
+			})
+
+			requirements = append(requirements, packit.BuildPlanRequirement{
+				Name: "dotnet-sdk",
+				Metadata: map[string]interface{}{
+					"version":        getSDKVersion(version),
+					"version-source": filepath.Base(projectFile),
+					"launch":         true,
+				},
+			})
+
+			aspNetIsRequired, err := projectParser.ASPNetIsRequired(projectFile)
+			if err != nil {
+				return packit.DetectResult{}, err
+			}
+
+			if aspNetIsRequired {
+				requirements = append(requirements, packit.BuildPlanRequirement{
+					Name: "dotnet-aspnetcore",
+					Metadata: map[string]interface{}{
+						"version":        version,
+						"version-source": filepath.Base(projectFile),
+						"launch":         true,
 					},
-					{
-						Name: "icu",
-						Metadata: map[string]interface{}{
-							"launch": true,
-						},
+				})
+			}
+
+			nodeIsRequired, err := projectParser.NodeIsRequired(projectFile)
+			if err != nil {
+				return packit.DetectResult{}, err
+			}
+
+			if nodeIsRequired {
+				requirements = append(requirements, packit.BuildPlanRequirement{
+					Name: "node",
+					Metadata: map[string]interface{}{
+						"version-source": filepath.Base(projectFile),
+						"launch":         true,
 					},
-				},
+				})
+			}
+		}
+
+		return packit.DetectResult{
+			Plan: packit.BuildPlan{
+				Requires: requirements,
 			},
 		}, nil
 	}
 }
+
+func getSDKVersion(version string) string {
+	if version == "" {
+		return "*"
+	}
+	pieces := strings.SplitN(version, ".", 3)
+	if len(pieces) < 3 {
+		pieces = append(pieces, "*")
+	}
+
+	var parts []string
+	for i, part := range pieces {
+		if i+1 == len(pieces) {
+			part = "*"
+		}
+
+		parts = append(parts, part)
+
+		if part == "*" {
+			break
+		}
+	}
+
+	return strings.Join(parts, ".")
+}
```

Comment:
Do you actually need to determine a SDK version here? Does this SDK version have any bearing on the final version selected and should it considering there is only 1 SDK for any given runtime and we install the SDK version based on the runtime we installed earlier?

Reply:
SDK version has no bearing on the selected runtime version, it's the other way around.
(1) App code parsers determine runtime version from app contents.
(2) The compatible sdk (containing cli) version is then found from this runtime version, so that the build phase can set the start command of `dotnet <app.dll>` for FDDs, and be assured that the right version of `dotnet` is used here.

Reply:
The version specified here is always just the major and minor of the runtime followed by a wildcard patch. Its just telling dotnet-core-sdk that we want an SDK from that major/minor line. Not sure I follow why that would be an issue.

Reply:
I just don't think that the major and minor have any bearing on the version of the SDK that should be installed. Let's say the major minor here is `2.1` but the runtime rolls forward to like `2.3.6` we should be installing the compatible SDK version for `2.3.6` and only checking to see if that violates `global.json`.

Reply:
Wouldn't the SDK choice just roll forward too?